lib/ansible/module_utils/common/locale.py

```diff
@@ -7,43 +7,51 @@
 from ansible.module_utils._text import to_native
 
 
-def get_best_parsable_locale(module, preferences=None):
+def get_best_parsable_locale(module, preferences=None, raise_on_locale=False):
     '''
         Attempts to return the best possible locale for parsing output in English
         useful for scraping output with i18n tools. When this raises an exception
         and the caller wants to continue, it should use the 'C' locale.
 
         :param module: an AnsibleModule instance
         :param preferences: A list of preferred locales, in order of preference
+        :param raise_on_locale: boolean that determines if we raise exception or not
+                                due to locale CLI issues
         :returns: The first matched preferred locale or 'C' which is the default
     '''
 
-    locale = module.get_bin_path("locale")
-    if not locale:
-        # not using required=true as that forces fail_json
-        raise RuntimeWarning("Could not find 'locale' tool")
-
-    available = []
     found = 'C'  # default posix, its ascii but always there
+    try:
+        locale = module.get_bin_path("locale")
```

Comment:
To avoid having to pass `module` and allow for using this function in places where `module` is not available (yet) would it be possible to use Python's `locale` module for this?

Reply:
If we must use `get_bin_path` we could instead import it from it's source instead: `lib/ansible/module_utils/common/process.py`

Reply:
We would still need `AnsibleModule`'s `run_command` to run the bin.

Reply:
Sorry that wasn't clear from the line I commented on.

Reply:
we cannot use python's `locale` since it reports 'possible locales' and not usable locales

and yes, we can switch `get_bin_path` but `run_command` would still require the `module` as a parameter